### PR TITLE
💅 Pretty maps and arrays

### PR DESCRIPTION
**Pull request by rowanseymour**
Addresses #401

### CODE REVIEW

File: cmd/docgen/renderers.go [hunk 1]
```diff
@@ -258,7 +259,8 @@ func checkExample(session flows.Session, line string) error {
 	}
 
 	test := strings.TrimSpace(pieces[0])
-	expected := strings.Replace(strings.TrimSpace(pieces[1]), "\\n", "\n", -1)
+	expected := strings.Replace(strings.TrimSpace(pieces[1]), `\n`, "\n", -1)
+	expected = strings.Replace(expected, `\x20`, " ", -1)
```
Comment (rowanseymour) [inline cmd/docgen/renderers.go]: need a way of encoding a trailing space in an example output

**Comment by ericnewcomer** [comment]
+1 for emoji

File: flows/engine/session_test.go [hunk 2]
```diff
@@ -79,12 +79,12 @@ func TestEvaluateTemplateAsString(t *testing.T) {
...
 		{"@input", "Hi there\nhttp://s3.amazon.com/bucket/test.jpg\nhttp://s3.amazon.com/bucket/test.mp3", ""},
 		{"@input.text", "Hi there", ""},
-		{"@input.attachments", `["http://s3.amazon.com/bucket/test.jpg","http://s3.amazon.com/bucket/test.mp3"]`, ""},
+		{"@input.attachments", `http://s3.amazon.com/bucket/test.jpg, http://s3.amazon.com/bucket/test.mp3`, ""},
 		{"@(input.attachments[0])", "http://s3.amazon.com/bucket/test.jpg", ""},
 		{"@input.created_on", "2017-12-31T11:35:10.035757-02:00", ""},
 		{"@input.channel.name", "My Android Phone", ""},
 
-		{"@results", `{"2factor":"34634624463525","favorite_color":"red","phone_number":"+12344563452"}`, ""},
+		{"@results", "2Factor: 34634624463525\nFavorite Color: red\nPhone Number: +12344563452", ""},
```
Comment (rowanseymour) [inline flows/engine/session_test.go]: note this is using result name as key

**Comment by nicpottier** [comment]
So urns are displaying their `format_urn` version ya? not their `string(urn)` version? So in this case stringed version of the higher element is showing formatted of sub-elements. Maybe that's ok just want to voice it.

**Comment by rowanseymour** [comment]
`text(x)` will always be what `x` stringifies to.. one is just explicit, other implicit

Currently for a URN that looks like...

```
@(contact.urns[0]) == "tel:+1234567890"
@(text(contact.urns[0])) == "tel:+1234567890"
@(format_urn(contact.urns[0])) == "(234) 567-890"
```

**Comment by nicpottier** [comment]
But `@(contact)` -> something with urns that are formatted right?

**Comment by nicpottier** [comment]
What I'm doing a poor job explaining is I'm wondering if having `text()` on a meta item turn inner items into non `text` is right.

IE, I think I would expect `text` to call `text` on everything inner.

Just as I would expect a `format()` to call `format()` on everything inner.

**Comment by rowanseymour** [comment]
In the case of maps and arrays, `text()` is a concatenation of `text()` of each item in that thing

For complex objects there's no concrete rule, `text(contact)` is the name, `text(run)` is the UUID, `text(results.response_1)` is the value.

Where are you seeing an inner item turned into non `text`?

Then we currently have a limited set of `format_` methods for humanzing/customizing stringification of some primitive types but we haven't generalised that to all types.

**Comment by nicpottier** [comment]
Ok cool.. I think we're on the same page then. For some reason I thought I saw somewhere where URNs were being turned into displays but I must have misread the HTML diff there.

**Pull request merged**